File: BE1/Models/Notification.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BE1.Models
{
    public class Notification
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = string.Empty;

        [BsonElement("recipientId")]
        public string RecipientId { get; set; } = string.Empty;

        [BsonElement("actorId")]
        public string ActorId { get; set; } = string.Empty;

        [BsonElement("type")]
        public string Type { get; set; } = string.Empty; // "follow", "like", "comment", "mention"

        [BsonElement("targetType")]
        public string TargetType { get; set; } = string.Empty; // "post", "comment", "user"

        [BsonElement("targetId")]
        public string TargetId { get; set; } = string.Empty;

        [BsonElement("message")]
        public string Message { get; set; } = string.Empty;

        [BsonElement("isRead")]
        public bool IsRead { get; set; } = false;

        [BsonElement("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: BE2/src/models/Notification.js
const mongoose = require('mongoose');

const notificationSchema = new mongoose.Schema({
  recipientId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: 'User', 
    required: true 
    // Người nhận thông báo (VD: Chủ bài viết)
  },
  senderId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: 'User',
    default: null
    
  },
  type: { 
    type: String, 
    enum: ['like', 'comment', 'follow', 'system'], 
    required: true 
   
  },
  postId: { 
    type: mongoose.Schema.Types.ObjectId, 
    ref: 'Post',
    default: null
    // Dẫn link tới bài viết (nếu có)
  },
  message: { 
    type: String, 
    required: true 
    
  },
  isRead: { 
    type: Boolean, 
    default: false 
    // Trạng thái Đã đọc / Chưa đọc
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Notification', notificationSchema);